src/widgets/helper/ChannelView.cpp

```diff
@@ -1786,8 +1789,48 @@ void ChannelView::leaveEvent(QEvent * /*event*/)
     this->unpause(PauseReason::Mouse);
 }
 
+bool ChannelView::event(QEvent *event)
+{
+     if (event->type() == QEvent::Gesture)
+     {
+         return gestureEvent(dynamic_cast<QGestureEvent*>(event));
+     }
+
+    return BaseWidget::event(event);
+}
+
+bool ChannelView::gestureEvent(const QGestureEvent *event)
+{
+     if (QGesture *pan = event->gesture(Qt::PanGesture))
+     {
+         auto const *gesture = dynamic_cast<QPanGesture *>(pan);
+
+         switch (gesture->state())
+         {
+             case Qt::GestureStarted:
+             case Qt::GestureUpdated:
+                 this->isPanning_ = true;
+                 this->clearSelection();
+                 break;
+             default:
+                 this->isPanning_ = false;
```

Comment:
I assume this is here to handle `Qt::GestureFinished` or `Qt::GestureCancelled` - I'd rather we handle those two states explicitly

Reply:
Yes these should be clearer now.